scripts/build-mimalloc.sh

```diff
@@ -25,4 +25,4 @@ cmake "${CMAKE_FLAGS[@]}" . \
 
 ninja
 
-cp libmimalloc.a $BUN_DEPS_OUT_DIR/libmimalloc.a
+cp libmimalloc.a $BUN_DEPS_OUT_DIR/libmimalloc.o
```

Comment:
A static library file is quite different from an object file ... 
The script should: 
+ generate an object file
+ the object file must be copied to the target directory 
if the object file is expected to be reused
Or
+ the current archive file must be properly handled by the buildsystem

Reply:
Hey jerome,

Thank you for the comment, I understand that an object file is different from a static library. In this context .o is not meant represent an object file rather a static library. I understand that this can cause confusion and is not ideal but unfortunately due how the CMake file is written for now, this is the solution. Note this is a hotfix not a permanent one. Look at the cmake file at

`
if(WIN32)
        target_link_libraries(${bun} PRIVATE "${BUN_DEPS_OUT_DIR}/mimalloc-static.lib")
    elseif(APPLE)
        # https://github.com/microsoft/mimalloc/issues/512
        # Linking mimalloc via object file on macOS x64 can cause heap corruption
        target_link_libraries(${bun} PRIVATE "${BUN_DEPS_OUT_DIR}/libmimalloc.a")
    else()
        target_link_libraries(${bun} PRIVATE "${BUN_DEPS_OUT_DIR}/libmimalloc.o")
    endif()
`

Reply:
Given the copied CMakefile content, the current fix will not work in all cases because object files can also be generated: 
+ `[ -f libmimalloc.a ] && cp ...`
+` [ -f libmimalloc.o ] && cp ...`

Reply:
It is my understanding that object files will not be generated due to the `DMI_BUILD_STATIC=ON` flag. However a second cp command as `cp libmimalloc.a $BUN_DEPS_OUT_DIR/libmimalloc.a` would have helped the mac builds. Eitherway this PR is moot now due to the new script that was just pushed

Reply:
Fair enough. Let's test the new build system then.